### PR TITLE
fix(date-input): add locale props from date-fns

### PR DESCRIPTION
**Pull request by philibea**
## Type

- Enhancement

### Summarise concisely:

Add possibilty to register locale of date-fns

Add locale props. it's add control

#### The following changes where made:

1. Add register of local with currentLocale
2. Add storybook with date-fns
3. Add Storybook with i18n  hook from scaleway-lib

## Relevant logs and/or screenshots

French Locale
Capture d’écran 2021-04-21 à 16 05 37

Spanish Locale

Capture d’écran 2021-04-21 à 16 05 51

Russian Locale
Capture d’écran 2021-04-21 à 16 06 01

I18n Provider
Capture d’écran 2021-04-21 à 16 06 23

### CODE REVIEW

File: src/components/DateInput/index.js [hunk 1]
```diff
@@ -224,6 +226,7 @@ DateInput.propTypes = {
   error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
   format: PropTypes.func,
   label: PropTypes.string,
+  locale: PropTypes.shape({}),
```
Comment (chambo-e) [inline src/components/DateInput/index.js:229]: Should be string here :)

Comment (philibea) [reply src/components/DateInput/index.js:235]: Locale is date-fns locale object and not currentLocale string

Comment (chambo-e) [reply src/components/DateInput/index.js:235]: Gotcha thanks, just wondering if we could use `locale.code` instead of the `currentLocale` props ?

Comment (philibea) [reply src/components/DateInput/index.js]: Yes you right useless props in this case.

**Review by vincentaudebert — APPROVED** [review]

**Review by chambo-e — APPROVED** [review]

**Pull request merged**